components/ai_chat/resources/untrusted_conversation_frame/components/markdown_renderer/index.tsx

```diff
@@ -156,6 +165,10 @@ interface MarkdownRendererProps {
 export default function MarkdownRenderer(mainProps: MarkdownRendererProps) {
   const lastElementRef = React.useRef<HastElement | undefined>()
 
+  // For table header tracking
+  const tableHeadersRef = React.useRef<string[]>([])
+  const rowIndexRef = React.useRef<number>(0)
```

Comment:
nit: I feel like this is actually a column index

Hey, so I can't think of a better way of doing this than what you've got, but maybe let's try and factor it into a function so its at least a bit more encapsulated.

What do you think of something like this:

```tsx
function buildTableRenderer() {
    const tableHeaders = []
    let columnIndex = 0

    return {
        table(props) {
            tableHeaders.length = 0
            // other table stuff
        },
        tr(props) {
            columnIndex = 0
            // other row stuff
        },
        td(props) {
            columnIndex++
        }
    }
}

// and then use it by spreading into `components`

components={{
    p: (...) => {...},
    ...buildTableRenderer()
}}

Reply:
Separated out into a function!